yaksh/static/yaksh/js/requesthandler.js

```diff
@@ -0,0 +1,139 @@
+request_status = "initial"
+count = 0;
+checker = null
+function submitRequest(){
+    document.forms["code"].submit();
+}
+
+function check_state(state, uid) {
+    if ((state == "running" || state == "not started") && count < 7) {
+        count++;
+        setTimeout(function() {get_result(uid);}, 2000);
+    } else if (state == "unknown") {
+        request_status = "initial";
+        count = 0;
+        notify("Request timeout. Try again later");
+        clearInterval(checker);
+        unlock_screen();
+    } else {
+        request_status = "initial";
+        count = 0;
+        notify("Please try after few minutes");
+        clearInterval(checker);
+        unlock_screen();
+    }
+}
+
+function notify(text) {
+    var $notice = document.getElementById("notification");
+    $notice.classList.add("alert");
+    $notice.classList.add("alert-success");
+    $notice.innerHTML = text;
+}
+
+function lock_screen() {
+    document.getElementById("ontop").style.display = "block";
+}
+
+function unlock_screen() {
+    document.getElementById("ontop").style.display = "none";
+}
+
+function check_lock_screen() {
+    var $ontop_div = document.getElementById("ontop");
+    if ($ontop_div.style.display == "block") {
+        $ontop_div.style.display = "none";
+    }
+}
+
+function get_result(uid){
+    var url = "/exam/get_results/"+uid+"/";
+    ajax_call(url, "GET", "html", null, uid)
+}
+
+function ajax_call(url, method_type, data_type, data, uid) {
+    $.ajax({
+        method: method_type,
+        url: url,
+        data: data,
+        dataType: data_type, // Your server can response html, json, xml format.
+        success: function(data, status, xhr) {
+            content_type = xhr.getResponseHeader("content-type");
+            if(content_type.indexOf("text/html") !== -1) {
+                if( method_type === "POST") {
+                    request_status = "initial";
+                    count = 0;
+                }
+                clearInterval(checker);
```

Comment:
This is very hacky that you create an interval checker and then clear it.  The implementation here seems rather hacky and unclear.  One year from now, no one will be able to figure out what is going on inside this JS code.  Please make it much simpler than it currently is.